test/cluster/mzcompose.py

```diff
@@ -4050,3 +4050,88 @@ def collect_read_frontiers() -> dict[str, int]:
     # Drop the cluster to cancel the subscribe.
     c.sql("DROP CLUSTER test CASCADE")
     subscribe_thread.join()
+
+
+def workflow_test_mz_introspection_cluster_compat(
+    c: Composition, parser: WorkflowArgumentParser
+) -> None:
+    """
+    Tests that usages of the `mz_introspection` cluster and the
+    `auto_route_introspection_queries` variable, which both have been renamed,
+    are automatically translated to the new names.
+    """
+
+    c.down(destroy_volumes=True)
+    c.up("materialized")
+
+    with c.override(
+        Testdrive(no_reset=True),
+    ):
+        c.up("testdrive", persistent=True)
+        c.up("materialized")
+
+        # Setting variables through `SET <variable>`.
+        c.testdrive(
+            dedent(
+                """
+                > SHOW cluster
+                quickstart
+
+                > SET cluster = mz_introspection
+                > SHOW cluster
+                mz_catalog_server
+
+                > SHOW auto_route_catalog_queries
+                on
+
+                > SET auto_route_introspection_queries = off
+                > SHOW auto_route_catalog_queries
+                off
+
+                > RESET cluster
+                > SHOW cluster
+                quickstart
+
+                > RESET auto_route_introspection_queries
+                > SHOW auto_route_catalog_queries
+                on
+                """
+            )
+        )
+
+        # Setting variables through `ALTER ROLE`.
+        c.sql(
+            """
+            ALTER ROLE materialize SET cluster = mz_introspection;
+            ALTER ROLE materialize SET auto_route_introspection_queries = off;
+            """
+        )
+        c.testdrive(
+            dedent(
+                """
+                > SHOW cluster
+                mz_catalog_server
+                > SHOW auto_route_catalog_queries
+                off
+                """
+            )
+        )
+        c.sql(
+            """
+            ALTER ROLE materialize RESET cluster;
+            ALTER ROLE materialize RESET auto_route_introspection_queries;
+            """
+        )
+        c.testdrive(
+            dedent(
+                """
+                > SHOW cluster
+                quickstart
+                > SHOW auto_route_catalog_queries
+                on
+                """
+            )
+        )
+
+        # Setting variables through the connection string.
+        # FIXME: how to do this?
```

Comment:
@nrainer-materialize Any ideas? I'd need to be able to specify a psql connection string with `--options`, which I'm not sure I can do with the `Composition` API.

Alternatively, I'd be happy to move to another framework (slt, testdrive) but I don't think there is one that allows me to test all three ways of specifying a cluster?

Reply:
What options would you like to set?

Reply:
Does it help you to call `c.sql_connection()` or `c.sql_cursor()` to create a connection and extend that method if necessary?

Reply:
You could check if it works with `psycopg2`, which we also use as a dependency:
https://github.com/MaterializeInc/materialize/blob/4dc75293172aca59c5ffcdfe858c761a1e2078d3/test/lang/python/smoketest.py#L85

Reply:
Thanks, that worked!